packages/flutter_tools/templates/create/test/widget_test.dart.tmpl

```diff
@@ -0,0 +1,44 @@
+// This is a basic Flutter widget test.
+// To perform an interaction with a widget in your test, use the WidgetTester utility that Flutter
+// provides. For example, you can send tap and scroll gestures. You can also use WidgetTester to
+// find child widgets in the widget tree, read text, and verify that the values of widget properties
+// are correct.
+
+import 'package:flutter/material.dart';
+import 'package:flutter_test/flutter_test.dart';
+
+void main() {
+  testWidgets('my first widget test', (WidgetTester tester) async {
+    // You can use keys to locate the widget you need to test
+    var sliderKey = new UniqueKey();
+    var value = 0.0;
+
+    // Tells the tester to build a UI based on the widget tree passed to it
+    await tester.pumpWidget(
+      new StatefulBuilder(
+        builder: (BuildContext context, StateSetter setState) {
+          return new Material(
+            child: new Center(
+              child: new Slider(
+                key: sliderKey,
+                value: value,
+                onChanged: (double newValue) {
+                  setState(() {
+                    value = newValue;
+                  });
+                },
+              ),
+            ),
+          );
+        },
+      ),
+    );
+    expect(value, 0.0);
+
+    // Taps on the widget found by key
+    await tester.tap(find.byKey(sliderKey));
+
+    // Verifies that the widget updated the value correctly
+    expect(value, 0.5);
+  });
```

Comment:
@Hixie : looking at improving the sample now.  Are you OK with this landing and us continuing a separate PR w/ improvements?  I suspect that we may want a bit of back and forth (and possibly other input).  For example, should we bake a unique key into the sample app itself?  Are there specific testing idioms we want to encourage?